CMakeLists.txt

```diff
@@ -46,21 +44,35 @@ endif()
 # Add base libs (emlib, CMSIS, device files)
 include(${BASE_LOCATION}/toolchain/efm32-base.cmake)
 
+add_subdirectory(cmsis)
+add_subdirectory(emlib)
+add_subdirectory(device)
+add_subdirectory(protocol)
+add_subdirectory(hardware)
+
+# this is normally set in efm32-base.cmake, but libraries may modify it so set
+# it after libraries/subdirectories have been added
+set(CMAKE_EXE_LINKER_FLAGS "${COMMON_DEFINITIONS} -Xlinker -T${LINKER_SCRIPT} -Wl,-Map=${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_PROJECT_NAME}.map -Wl,--gc-sections -Wl,-v")
```

Comment:
this ties in with #10 i think, fine here for now but could be moved into linker config function when we get there